### ISSUE TITLE
During random sleep, configuration file changes are not processed

### ISSUE THREAD

**Issue by tiedotguy** [adferrand/dnsrobocert#263]
When the `_renew_job` function is invoked, it effectively blocks execution of the main loop, meaning configuration files aren't processed, possibly for several hours.

One approach to address this would be for `_renew_job` to spin off a new job and return immediately, something like...

```
def _renew_job(config_path: str, directory_path: str):
    random_delay_seconds = 21600  # Random delay up to 12 hours
    wait_time = int(random() * random_delay_seconds)
    LOGGER.info("Automated execution: renew certificates if needed.")
    LOGGER.info(f"Random wait for this execution: {wait_time} seconds")

    schedule.every(wait_time).seconds.do(_actual_renew_job, config_path=config_path, directory_path=directory_path)

def _actual_renew_job(config_path: str, directory_path: str):
    certbot.renew(config_path, directory_path)
    return schedule.CancelJob  # don't repeat the job
```

**Comment by adferrand** [comment]
Hello! It is true, I am starting to work on that, thanks for the suggestion!

**Comment by adferrand** [comment]
I think this issue has been solved since version 3.9.0, by running renewal jobs in a dedicated thread. If this is not sufficient, do not hesitate to reopen the issue!
